src/cpu.rs

```diff
@@ -33,7 +33,9 @@ pub(crate) fn features() -> Features {
             target_arch = "x86",
             target_arch = "x86_64"
         ),
-        not(target_os = "ios")
+        not(
+            any(target_os = "ios",
+                target_os = "macos"))
```

Comment:
AFAICT, this would have disabled CPU feature detection on x86-64 Macs, so I didn't take this change.

Also, when I was trying to understand your PR, I noticed the bug that PR #1100 fixes. (That bug is just a performance bug, not a correctness bug, but it is a pretty severe performance bug.)